### ISSUE TITLE
How do we predict data without targets in SVM Kernels?

### ISSUE THREAD

**Issue by zhedongzheng** [nfmcclure/tensorflow_cookbook#72]
Hello, thanks for the excellent work.

I am confused in `04_svm_kernels.ipynb` of Chapter 4 Support Vector Machine, where we always require `y_target` in prediction because of dual calculation. However, in real world practice, if we only have X_test to predict Y_test, how do we get prediction without the labels?

Do you have any insight about this?

Thanks

**Comment by nfmcclure** [comment]
Hi @zhedongzheng ,
Thanks for the comment!  Unfortunately, the SVM algorithm needs the y-targets of the training set for training. In fact, there are specific y-targets from the training set that make up the so-called "support vectors" of the separation boundary that determines the classification.  So in order to make predictions on a test set with no y-targets, the training set (with the y-targets) is needed.  This is the same with most classification algorithms, except the need for the training set is more specific in the SVM algorithms.

If you are done training a decision boundary and want to forget the support vectors, there may exist a way to approximate the boundary with a set of points or curve parameters, but then we lose the notion of a measurable loss with a test set.

I'm going to close this for now, but feel free to open it if I misunderstood your point or more clarification is needed. Thanks again!
